backend/inspirehep/records/api/base.py

```diff
@@ -553,3 +553,10 @@ def redirected_record_ref(self):
         if self.get("deleted") and pid.status == PIDStatus.REDIRECTED:
             redirection = InspireRedirect.get_redirect(pid)
             return get_ref_from_pid(redirection.pid_type, redirection.pid_value)
+
+    def check_if_record_was_deleted_by_merge(self):
+        current_deleted_records = get_value(self, "deleted_records.$ref", [])
+        if current_deleted_records != get_value(
+            self._previous_version, "deleted_records.$ref", []
+        ):
+            return current_deleted_records.pop()
```

Comment:
:warning: you might be mutating the record here (I don't believe `get_value` makes a copy, I'm not entirely sure)